src/main/java/com/icthh/xm/uaa/security/oauth2/token/OAuth2XmAccessTokenJackson2Serializer.java

```diff
@@ -0,0 +1,64 @@
+package com.icthh.xm.uaa.security.oauth2.token;
+
+import com.fasterxml.jackson.core.JsonGenerationException;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import java.io.IOException;
+import java.util.Date;
+import java.util.Map;
+import java.util.Set;
+import org.springframework.security.oauth2.common.ExpiringOAuth2RefreshToken;
+import org.springframework.security.oauth2.common.OAuth2AccessToken;
+import org.springframework.security.oauth2.common.OAuth2RefreshToken;
+import org.springframework.util.Assert;
+
+/**
+ * Provides the ability to serialize an {@link XmOAuth2AccessToken} with jackson2 by implementing {@link com.fasterxml.jackson.databind.JsonDeserializer}.
+ * Refer to {@link org.springframework.security.oauth2.common.OAuth2AccessTokenJackson2Serializer} to see the original implementation
+ */
+public final class OAuth2XmAccessTokenJackson2Serializer extends StdSerializer<OAuth2AccessToken> {
+
+    private static final String REFRESH_TOKEN_EXPIRES_IN = "refresh_token_expires_in";
+
+    public OAuth2XmAccessTokenJackson2Serializer() {
+        super(OAuth2AccessToken.class);
+    }
+
+    @Override
+    public void serialize(OAuth2AccessToken token, JsonGenerator jgen, SerializerProvider provider)
```

Comment:
can we override other part of flow to add additional field, serializer don't look like right solution